layers.go

```diff
@@ -2529,6 +2529,7 @@ func (r *layerStore) applyDiffFromStagingDirectory(id string, diffOutput *driver
 	layer.GIDs = diffOutput.GIDs
 	updateDigestMap(&r.byuncompressedsum, layer.UncompressedDigest, diffOutput.UncompressedDigest, layer.ID)
 	layer.UncompressedDigest = diffOutput.UncompressedDigest
+	layer.CompressedDigest = diffOutput.CompressedDigest
```

Comment:
- This might overwrite a valid `CompressedDigest` value (hypothetically, from a template layer) with `""`
- If this field is modified, a corresponding `updateDigestMap` call needs to happen as well.

I think the other parts of this PR are clearly valuable; this one doesn’t really hurt but seems unnecessary for the purpose. OTOH handling the `DriverWithDifferOutput` fields consistently _is_ somewhat valuable.

Reply:
thanks, I'll fix it and call `updateDigestMap`